sql/hive/src/test/resources/golden/decimal_4-6-693c2e345731f9b2b547c3b75218458e

```diff
@@ -1,38 +1,38 @@
 NULL	0
--1234567890.123456789	-1234567890
--4400	4400
--1255.49	-1255
--1.122	-11
--1.12	-1
--1.12	-1
--0.333	0
--0.33	0
--0.3	0
-0	0
-0	0
-0	0
-0.01	0
-0.02	0
-0.1	0
-0.2	0
-0.3	0
-0.33	0
-0.333	0
+-1234567890.1234567890000000000000000	-1234567890
+-4400.0000000000000000000000000	4400
+-1255.4900000000000000000000000	-1255
+-1.1220000000000000000000000	-11
+-1.1200000000000000000000000	-1
+-1.1200000000000000000000000	-1
+-0.3330000000000000000000000	0
+-0.3300000000000000000000000	0
+-0.3000000000000000000000000	0
+0.0000000000000000000000000	0
```

Comment:
The case '0' is the same with the current Hive?

Reply:
Yes. https://github.com/apache/hive/blob/release-2.3.5-rc0/ql/src/test/results/clientpositive/decimal_4.q.out#L69

```scala
scala> val d1 = java.math.BigDecimal.ZERO
d1: java.math.BigDecimal = 0

scala> val d2 = java.math.BigDecimal.ZERO.setScale(8)
d2: java.math.BigDecimal = 0E-8

scala> d1.compareTo(d2)
res0: Int = 0

scala> d1.toPlainString
res1: String = 0

scala> d2.toPlainString
res2: String = 0.00000000
```

Reply:
Thanks for the check!